visidata/loaders/grep.py

```diff
@@ -0,0 +1,137 @@
+                        continue
+                    json_obj = json.loads(L)
+                    if not isinstance(json_obj, dict):
+                        vd.fail(f'line does not hold a JSON object:  {L}')
+                    row = format_row(json_obj)
+                    if not row: #skip lines that do not contain match data
+                        continue
+                    yield row
+                except ValueError as e:
+                    if self.rows:   # if any rows have been added already
+                        e.stacktrace = stacktrace()
+                        yield TypedExceptionWrapper(json.loads, L, exception=e)  # an error on one line
+                    else:
+                        # If input is not JSON, parse it as output of 'grep -n':  file:line_no:text
+                        # If that does not parse, parse it as output of typical 'grep':  file:text
+                        with self.open_text_source() as fp:
+                            try:
+                                extract_line_no = True
+                                for L in fp:
+                                    L = L.rstrip(linesep)
+                                    sep1 = L.index(':')
+                                    if extract_line_no:
+                                        sep2 = L.find(':', sep1+1)
+                                        try:
+                                            if sep2 == -1: raise ValueError
+                                            line_no = int(L[sep1+1:sep2]) # may raise ValueError
+                                            if line_no < 1: raise ValueError
+                                            text = L[sep2+1:]
+                                        except ValueError: # if we can't find a line_no that is > 0, with a separator after it
+                                            extract_line_no = False
+                                            line_no = None
+                                            text = L[sep1+1:]
+                                    else:
+                                        text = L[sep1+1:]
+                                    yield AttrDict({'file':    L[:sep1],
+                                                    'line_no': line_no,
+                                                    'text':    text})
+                            except ValueError:
+                                vd.fail('file is not grep output')
+                        break
+
+    def afterLoad(self):
+        if self.nRows == 0:
+            vd.status('no grep results found in input data')
+
+@GrepSheet.api
+def sysopen_row(sheet, row):
+    '''Open the file in an editor at the specific line.'''
+    if sheet.nRows == 0: return
+    try:
+        given = row.file
+        if vd.options.grep_base_dir and not os.path.isabs(given):
+            given = vd.options.grep_base_dir + os.sep + row.file
+        p = Path(given)
+    except:
```

Comment:
Never use a bare `except:` in Python.  At minimum, use `except Exception:`, as almost all exceptions inherit from Exception.  This is a general rule for Python, but in fact is important for VisiData, as its `EscapeException` inherits from Python's `BaseException` (which is the true base class for all exceptions), and is used when the user cancels an operation with Ctrl+C.  So anywhere a bare `except:` is used in VisiData, may lead to a situation where the user cannot cancel an operation.

Reply:
Oh, interesting! Okay, I've fixed it to be more specific.